File: unix/boot/spp/rpp/rpprat/mapid.r

include defs

# MAPID -- Map a long identifier.  The new identifier is formed by
# concatenating the first MAXIDLENGTH-1 characters and the last character.


subroutine mapid (name)

character name(MAXTOK)
integer	i

	for (i=1;  name(i) != EOS;  i=i+1)
	    ;
	if (i-1 > MAXIDLENGTH) {
	    name(MAXIDLENGTH) = name(i-1)
	    name(MAXIDLENGTH+1) = EOS
	}
end

File: iraf/unix/boot/spp/rpp/rppfor/mapid.f
      subroutine mapid (name)
      integer name(100)
      integer i
      i=1
23000 if (.not.(name(i) .ne. -2))goto 23002
23001 i=i+1
      goto 23000
23002 continue
      if (.not.(i-1 .gt. 6))goto 23003
      name(6) = name(i-1)
      name(6+1) = -2
23003 continue
      end
